### PR TITLE
Use a service account to access other API's

### PR DESCRIPTION
**Pull request by Matt-Guest**
### What

Added service account token to HTTP requests going out.

On startup the service account token is check, to see if its valid. If this check is
failed the app shall exit.

### How to review

Review the code, run the unit tests and run some manual tests

### Who can review

anyone

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -27,7 +27,8 @@ setup as environment variables.
 | FILTER_API_URL        | http://localhost:22100               | Filter api URL
 | FILTER_API_AUTH_TOKEN | FD0108EA-825D-411C-9B1D-41EF7727F465 | Secret token to use the Filter api
 | DATASET_API_URL       | http://localhost:22000               | Dataset api URL
-| DATASET_API_AUTH_TOKEN| FD0108EA-825D-411C-9B1D-41EF7727F465 | Secret token to use the Dataset api
+| SERVICE_AUTH_TOKEN    | FD0108EA-825D-411C-9B1D-41EF7727F465 | Secret token to use the Dataset api
```
Comment (nshumoogum) [inline README.md]: Default value needs updating for `SERVICE_AUTH_TOKEN`

File: README.md [hunk 2]
```diff
@@ -27,7 +27,8 @@ setup as environment variables.
 | FILTER_API_URL        | http://localhost:22100               | Filter api URL
```
Comment (nshumoogum) [inline README.md]: If you could update the README.md to specify a need/dependency for the auth-api-stub

File: README.md (hunk 1 above)
Comment (CarlHembrough) [reply README.md]: Update default value and its comment

File: README.md (hunk 2 above)
Comment (CarlHembrough) [reply README.md]: Added another point to the getting started section about running the stub or zebedee

File: src/main/java/dp/api/authentication/AuthenicateOnStartUp.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+@Component
+public class AuthenicateOnStartUp implements ApplicationListener<ApplicationReadyEvent> {
+
+    private final static Logger LOGGER = LoggerFactory.getLogger(AuthenicateOnStartUp.class);
+
+    @Value("${ZEBEDEE_URL:}")
+    private String zebedee_url;
+
+    @Value("${SERVICE_AUTH_TOKEN:939616dc-7599-4ded-9a86-a9c66fbf98e0}")
```
Comment (nshumoogum) [inline src/main/java/dp/api/authentication/AuthenicateOnStartUp.java]: Update default value to `7049050e-5d55-440d-b461-319f8cdf6670`

File: README.md [hunk 4]
```diff
@@ -27,7 +28,8 @@ setup as environment variables.
...
 | FILTER_API_AUTH_TOKEN | FD0108EA-825D-411C-9B1D-41EF7727F465 | Secret token to use the Filter api
 | DATASET_API_URL       | http://localhost:22000               | Dataset api URL
-| DATASET_API_AUTH_TOKEN| FD0108EA-825D-411C-9B1D-41EF7727F465 | Secret token to use the Dataset api
+| SERVICE_AUTH_TOKEN    | 939616dc-7599-4ded-9a86-a9c66fbf98e0 | Service token to authenticate against Zebedee
```
Comment (nshumoogum) [inline README.md]: Update default value to `7049050e-5d55-440d-b461-319f8cdf6670`

File: src/main/java/dp/api/dataset/DatasetAPIClientImpl.java [hunk 5]
```diff
@@ -28,28 +27,25 @@
...
     @Value("${DATASET_API_URL:http://localhost:22000}")
     private String datasetAPIURL;
 
+    @Deprecated
     @Value("${DATASET_API_AUTH_TOKEN:FD0108EA-825D-411C-9B1D-41EF7727F465}")
     private String token;
 
-    @Autowired
-    private RestTemplate restTemplate;
+    @Value("${SERVICE_AUTH_TOKEN:939616dc-7599-4ded-9a86-a9c66fbf98e0}")
```
Comment (nshumoogum) [inline src/main/java/dp/api/dataset/DatasetAPIClientImpl.java]: Update default value to `7049050e-5d55-440d-b461-319f8cdf6670`

File: src/main/java/dp/api/filter/FilterAPIClient.java [hunk 6]
```diff
@@ -23,9 +24,13 @@
     @Value("${FILTER_API_URL:http://localhost:22100}")
     private String filterAPIURL;
 
+    @Deprecated
     @Value("${FILTER_API_AUTH_TOKEN:FD0108EA-825D-411C-9B1D-41EF7727F465}")
     private String token;
 
+    @Value("${SERVICE_AUTH_TOKEN:07B082D8-AA33-4213-A562-911FA019C570}")
```
Comment (nshumoogum) [inline src/main/java/dp/api/filter/FilterAPIClient.java]: Update default value to `7049050e-5d55-440d-b461-319f8cdf6670`

**Pull request merged**